featuretools/primitives/standard/datetime_transform_primitives.py

```diff
@@ -331,6 +334,72 @@ def is_leap_year(vals):
         return is_leap_year
 
 
+class IsLunchTime(TransformPrimitive):
+    """Determines if a datetime falls during lunch time (non-holiday weekday with hour=12)
+
+    Args:
+        country (str): Country to use for determining Holidays.
+            Default is 'US'. Should be one of the available countries here:
+            https://github.com/dr-prodigy/python-holidays#available-countries
+        include_weekend (bool): Include weekend dates as datetimes with valid
+            lunch times. Default is True.
+        include_holidays (bool): Include holiday dates as datetimes with valid
+            lunch times. Default is False.
+
+    Examples:
+        >>> from datetime import datetime
+        >>> dates = [datetime(2022, 6, 21, 12, 3, 3),
+        ...          datetime(2019, 1, 3, 4, 4, 4),
+        ...          datetime(2022, 1, 1, 12, 1, 2),
+        ...          np.nan]
+        >>> ilt = IsLunchTime()
+        >>> ilt(dates).tolist()
+        [True, False, False, False]
+    """
+
+    name = "is_lunch_time"
+    input_types = [ColumnSchema(logical_type=Datetime)]
+    return_type = ColumnSchema(logical_type=BooleanNullable)
+    compatibility = [Library.PANDAS, Library.DASK, Library.SPARK]
+    description_template = "whether {} falls during lunch time"
+
+    def __init__(self, country="US", include_weekends=True, include_holidays=False):
+        self.country = country
+        years_list = [1950 + x for x in range(150)]
+        self.federal_holidays = getattr(holidays, self.country)(years=years_list)
+        self.include_weekends = include_weekends
+        self.include_holidays = include_holidays
+        self.holidays_df = pd.DataFrame(
+            sorted(self.federal_holidays.items()), columns=["dates", "names"]
+        )
+        self.timestamp_dates = [pd.Timestamp(i) for i in self.holidays_df.dates]
+
+    def spark_mask(self, v):
+        lunch_mask = v.hour == 12
+        if not self.include_weekends:
+            lunch_mask = lunch_mask and (v.dayofweek < 5)
+        if not self.include_holidays:
+            holiday_mask = v.normalize() in self.timestamp_dates
+            lunch_mask = lunch_mask and not holiday_mask
+        return lunch_mask
+
+    def get_function(self):
+        def is_lunch_time(vals):
+            if isinstance(vals, pd.Series) or isinstance(vals, dd.dataframe.Series):
```

Comment:
I think we should flip the logic around here. The code in the first block should be our default, and we only do special handling for spark.  Also, maybe use our custom `is_instance` function for checking the type of the series.

```python
if is_instance(vals, ps, "Series"):
   # code for Spark series here
else:
    # default code for pandas and Dask here
```